pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -2534,18 +2536,41 @@ def _is_valid_return_type(match: TypeHintMatch, node: nodes.NodeNG) -> bool:
 
     if (
         match.check_return_type_inheritance
-        and isinstance(match.return_type, str)
+        and isinstance(match.return_type, (str, list))
         and isinstance(node, nodes.Name)
     ):
-        ancestor: nodes.ClassDef
-        for infer_node in node.infer():
-            if isinstance(infer_node, nodes.ClassDef):
-                if infer_node.name == match.return_type:
+        if isinstance(match.return_type, str):
+            valid_types = {match.return_type}
+        elif isinstance(match.return_type, list):
+            valid_types = {el for el in match.return_type if isinstance(el, str)}
+        else:
+            pass
```

Comment:
This will never be executed. `match.return_type` is either `str` or `list`.

Reply:
Done